src/Symfony/Bundle/SecurityBundle/DependencyInjection/Security/Factory/FormLoginFactory.php

```diff
@@ -22,6 +22,8 @@
  *
  * @author Fabien Potencier <fabien@symfony.com>
  * @author Johannes M. Schmitt <schmittjoh@gmail.com>
+ *
+ * @final
```

Comment:
This breaks, as this class is extended by `FormLoginLdapFactory`. The same applies to json login and http basic. Is there a way to hide the errors for Symfony's own mis-use, or do I need to make all DI factories `@internal` instead? cc @fabpot @nicolas-grekas

Reply:
Nope, no way around, this is really meant to mean `final`.

Reply:
So then... we must remove `@final` from `FormLoginFactory`? That's right after we decided that they *should* really be final. Or do we need to re-organize the sharing, like into a trait?

Reply:
Or we make them `@internal`. This adds further limitations on the use of these factories, but allows extending it within the `Symfony\\` namespace IIRC.

Reply:
Marking them as `@internal` sounds like a good idea.